lib/msf/core/post/file.rb

```diff
@@ -147,6 +154,8 @@ def directory?(path)
   def expand_path(path)
     if session.type == "meterpreter"
       return session.fs.file.expand_path(path)
+    elsif session.type == 'powershell'
+      return cmd_exec("(Resolve-Path \"#{path}\").Path")
```

Comment:
This does not look like it's working. I compared the output of a test case on Meterpreter and Powershell.

```
msf6 post(windows/gather/checkvm) > set SESSION 2
SESSION => 2
msf6 post(windows/gather/checkvm) > pry
[*] Starting Pry shell...
[*] You are in the "post/windows/gather/checkvm" module object

[1] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> session.type
=> "meterpreter"
[2] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> expand_path("%TMP%\\Foo\\Bar")
=> "C:\\Windows\\TEMP\\Foo\\Bar"
[3] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> 
msf6 post(windows/gather/checkvm) > set SESSION 1
SESSION => 1
msf6 post(windows/gather/checkvm) > pry
[*] Starting Pry shell...
[*] You are in the "post/windows/gather/checkvm" module object

[1] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> session.type
=> "powershell"
[2] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> expand_path("%TMP%\\Foo\\Bar")
=> ""
[3] pry(#<Msf::Modules::Post__Windows__Gather__Checkvm::MetasploitModule>)> 
```

Reply:
Does `"%TMP%\\Foo\\Bar"` exists on the target system? because `Resolve-Path` in powershell doesn't work for non-existant paths.

Reply:
But looks like meterpreter and shell are expanding the path even when the path doesn't exist, so to make it consistent how about using something like `[$ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath(".\nonexist\foo.txt"]`(https://stackoverflow.com/a/3040982)

Reply:
```
>> expand_path('/')
=> "C:\\"
>> expand_path('/path/does/not/exists')
=> "C:\\path\\does\\not\\exists"
>> expand_path('../path/does/not/exists')
=> "C:\\Users\\path\\does\\not\\exists"
```